sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -173,7 +171,13 @@ case class ScriptTransformation(
           }
           outputStream.close()
         }
-      }).start()
+      }, "Thread-ScriptTransformation-Feed").start()
+
+      // Consume the error stream from the pipeline, otherwise it will be blocked if
+      // the pipeline is full.
+      new RedirectThread(errorStream, // input stream from the pipeline
+        new CircularBuffer(),         // output to a circular buffer
```

Comment:
Shouldn't we hold on to this and print it if the forked program exits with a non-zero exit code?